src/System.Net.Security/src/System/Net/Security/SslStreamPal.OSX.cs

```diff
@@ -235,18 +267,17 @@ public static void VerifyPackageInfo()
 
                 if ((null == context) || context.IsInvalid)
                 {
-                    sslContext = new SafeDeleteSslContext(credential as SafeFreeSslCredentials, isServer);
+                    sslContext = new SafeDeleteSslContext(credential as SafeFreeSslCredentials, sslAuthenticationOptions);
                     context = sslContext;
 
-                    if (!string.IsNullOrEmpty(targetName))
+                    if (!string.IsNullOrEmpty(sslAuthenticationOptions.TargetHost))
                     {
-                        Debug.Assert(!isServer, "targetName should not be set for server-side handshakes");
-                        Interop.AppleCrypto.SslSetTargetName(sslContext.SslContext, targetName);
+                        Debug.Assert(!sslAuthenticationOptions.IsServer, "targetName should not be set for server-side handshakes");
+                        Interop.AppleCrypto.SslSetTargetName(sslContext.SslContext, sslAuthenticationOptions.TargetHost);
                     }
 
-                    if (remoteCertRequired)
+                    if (sslAuthenticationOptions.IsServer && sslAuthenticationOptions.RemoteCertRequired)
                     {
-                        Debug.Assert(isServer, "remoteCertRequired should not be set for client-side handshakes");
```

Comment:
I asked this question in the OpenSSL version, but I'll ask it again: why are we removing the assert?  If RemoteCertRequired can only be set when IsServer is true, then the assert is good (because we can remove the `&&` operation).  If there's a state where RemoteCertRequired can be true and IsServer is false, then what do we want that behavior to be?

Reply:
As explained previously.